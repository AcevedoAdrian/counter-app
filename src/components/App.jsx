import React from "react";
import PropTypes from 'prop-types'
const App = ({ saludo, despedida }) => {


  const arreglo = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
  const boleanos = false;
  const objetoPersona = {
    nombre: 'Juan',
    apellido: 'Perez',
    edad: 30
  }

  return (<>
    <h1>{saludo}</h1>
    {/* Lo que esta entre llave no imprime boleanos ya que los evaluda, los undefine tampoco se imprimen en estas llaves*/}
    <h2>{boleanos}</h2>
    {/* Itera sobre el arreay y cada elemento para a ser un string */}
    <h2>{arreglo}</h2>
    {/* No se puede imprimir objetos hay que parsearlos   */}
    {/* <h1>{objetoPersona}</h1> */}
    <pre> {JSON.stringify(objetoPersona, null, 3)}</pre>

    <p>My first aplication</p>
  </>);
};
App.propTypes = {
  saludo: PropTypes.string.isRequired,
  despedida: PropTypes.string
}

// Definimos las propiedades que va a recibir el componente con un valor por defecto, para no escribirlo en el componente en los parametros 
App.defaultProps = {
  despedida: 'Adios',
  saludo: 'Holiwis'
}
export default App;