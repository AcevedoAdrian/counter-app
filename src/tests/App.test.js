import React from 'react';
import { shallow } from 'enzyme';
// import { render } from '@testing-library/react';
import App from '../components/App';
import '@testing-library/jest-dom';
describe('Probando componentes de react <App />', () => {


  // esto es para usar con el test de componentes de react que viene con jest
  // test('Debe de mostrar el componente <App /> Hola, soy Goku', () => {

  //   const saludo = 'Hola, soy Goku';
  //   const { getByText } = render(<App saludo={saludo} />);
  //   // expect(getByText(saludo)).toBeInTheDocument();
  // })

  test('Debe de mostrar <App /> de manera correcta', () => {
    const saludo = 'Hola, soy Goku';
    const wrapper = shallow(<App saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Subtitulo debe tener el texto enviado por props', () => {
    const saludo = 'Hola, soy Goku';
    const despedida = 'Adios';

    const wrapper = shallow(<App saludo={saludo} despedida={despedida} />);
    const textoParrafo = wrapper.find('p').text();
    expect(textoParrafo).toBe(despedida);
  });
});