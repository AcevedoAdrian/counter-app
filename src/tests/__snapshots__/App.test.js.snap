// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componentes de react <App /> Debe de mostrar <App /> de manera correcta 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
    !!
  </h1>
  <p>
    Adios
  </p>
</Fragment>
`;
